### PR TITLE
[UNDER DISCUSSION] apps/examples/smartfs: Separate submenu for selecting SmartFs Applications

### PR DESCRIPTION
**Pull request by amogh-samsung**
- Add new submenu in Application Examples menuconfig to display the SmartFs Test App options

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

### CODE REVIEW

**Comment by amogh-samsung** [comment]
#5139 
@jeongarmy please also remove the repetitive Kconfig files for apps/examples/performance
Currently, Performance Test Apps have a sub folder "Performance-->" and the options are also displayed directly in the "Applications Configuration" menu, i.e. there is a duplication
@sunghan-chang moved the Performance Test Applications to a sub folder earlier

We would like to have a separate sub folder for SmartFs test applications, but that results in duplication because  of "-maxdepth 3" in directory.mk
This problem can be resolved by keeping "-maxdepth 2" and having Kconfig files in each of the subfolders (apps/examples/<subfolder>) which will then be picked up and allow creation of sub menus in the menuconfig display

**Comment by amogh-samsung** [comment]
Additionally, the restructuring of the application folders was not done for RS6_Tp1x or Ecode due to which we are unable to sync our new SmartFs test applications to these repositories.
Requesting @jeongarmy and @sunghan-chang to kindly propose a final solution to maintain uniformity in the sub-folder/sub-menu arrangement i.e. whether to continue with "-maxdepth 3" for auto inclusion or another way that allows creation of sub-menus without duplication.

**Comment by sunghan-chang** [comment]
@amogh-samsung Sorry for late reply, but my machine was broken so that I can't check it yet. Once @jeongarmy is back to office, let me talk with her.

File: apps/examples/smartfs/Kconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#
+# For a description of the syntax of this configuration file,
+# see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
+#
+# This file is autogenerated, do not edit.
```
Comment (sunghan-chang) [inline apps/examples/smartfs/Kconfig:5]: As this line says, this is auto-generated file. This looks strange. Let me check another way.

**Comment by amogh-samsung** [comment]
> @amogh-samsung Sorry for late reply, but my machine was broken so that I can't check it yet. Once @jeongarmy is back to office, let me talk with her.

Yes please, sure.
Let's have a uniform way for all sub folders of applications?
Personally, I would prefer separate Kconfigs for subfolders as that allows us to create sub menus in the menuconfig for applications

**Comment by jeongarmy** [comment]
@amogh-samsung I agree with your opinion. I think it is better to revert depth to 2 and add Kconfigs for submenu.
Could you please review my changes, #5522?

**Comment by amogh-samsung** [comment]
Issue closed with #5526

**Pull request closed without merging**